src/Models/ModelFactory.php

```diff
@@ -24,12 +24,62 @@ public static function create(BaseClient $client, stdClass $responseObject, stri
                                               ProductModel => static::resolveProductModelAttributes($value, $resource, $result->id),
                 default => $value,
             };
+        }
+
+        $defaultLocale = $resource->getClient()->getErgonodeApi()->connector->getLocale();
 
+        if ($resource instanceof ProductModel && isset($resource->translations)) {
+            static::resolveTranslationsOnProductResource($resource, $defaultLocale);
         }
 
         return $resource;
     }
 
+    private static function resolveTranslationsOnProductResource(mixed $resource, string $defaultLocale): mixed
+    {
+        $resource->translations["attributes"] = (object)[];
+
+        foreach ($resource->attributes as $field => $value) {
+            $resource->translations["attributes"]->{$field} = (object)[
+                $defaultLocale => $value,
+            ];
+        }
+
+        $api             = $resource->getClient()->getErgonodeApi();
+        $languagesClient = $api->languages();
+        $languages       = $languagesClient->all(true);
+
+        foreach ($languages as $language) {
+            if ($defaultLocale === $language->code) {
+                continue;
+            }
+
+            $resultInLanguage = $api->connector->get($api->products()->singleUri() .
```

Comment:
Dit ff in een string "" in plaats van met een punt ervoor.

$singleUri = api->products()->singleUri();
"$singleUri//inherited/$language->code"